nucypher/crypto/keystore.py

```diff
@@ -342,15 +342,27 @@ def restore(cls, words: str, password: str, keystore_dir: Optional[Path] = None)
         return keystore
 
     @classmethod
-    def generate(cls, password: str, keystore_dir: Optional[Path] = None, interactive: bool = True) -> 'Keystore':
+    def generate(
+            cls, password: str,
+            keystore_dir: Optional[Path] = None,
+            interactive: bool = True,
+            report_mnemonic: bool = False
+            ) -> 'Keystore':
         """Generate a new nucypher keystore for use with characters"""
+        if report_mnemonic and interactive:
```

Comment:
It seems like this same logic can be expressed as a simple if/else against a single boolean flag. Why use two separate inputs?

Reply:
I wanted to honor the existing interactive flag -- but never return the mnemonic.words when interactive was turned on since that seemed unnecessary and possibly(?) a bit loose with security.   If you think there is a good use case for doing both interactive and returning mnemonic.words, please let me know and i'll get rid of this functionality.

Reply:
I don't see a use case in doing both, however the same behavior you have here can be expressed without `report_mnemonic`. Instead If `interactive=False` just return `keystore, __words`.

Reply:
ok i've cleaned this up.  you are quite right.  i was originally doing it i think in part because i feared many tests would need to be changed.  I have found one just now that seems to error because of my change.  Working on that now:

pytest -s tests/integration/config/test_configuration_persistence.py

Reply:
ok.  i think that the failing test is now passing with my latest change to base.py on line 802.

`            self.__keystore, _ = Keystore.generate(password=password,
`

Reply:
uh oh - i think that this will or should break other tests... working on more changes...

Reply:
I fixed the remaining broken test.